core/date_api.php

```diff
@@ -280,3 +280,62 @@ function print_date_selection_set( $p_name, $p_format, $p_date = 0, $p_default_d
 	}
 }
 
+/**
+ * Convert php date time format string to moment.js compatible format
+ * moment.js is a popular date time handling JS library. It is used by datetime picker library
+ * see https://stackoverflow.com/questions/30186611/php-dateformat-to-moment-js-format
+ * Moment.js date time format: https://momentjs.com/docs/#/parsing/string-format/
+ * Php date time format: https://www.php.net/manual/en/function.date.php
+ * @param string $p_php_format     	Datetime php format
+ * @return string 					Date time format string understood by moment.js based components
+ * @access public
+ */
+function convert_php_to_momentjs_datetime_format( string $p_php_format ) {
+	$t_replacements = [
+		'A' => 'A',      # for the sake of escaping below
+		'a' => 'a',      # for the sake of escaping below
+		'B' => '',       # Swatch internet time (.beats), no equivalent
```

Comment:
In the unlikely event that someone would actually use this, shouldn't we trigger an exception or at least a warning, instead of silently dropping the format ?

I guess the same applies also to other PHP formats which do not have a direct corresponding value in momentjs (e.g. `L`, `t` etc)

Reply:
It does not look like you've addressed [my earlier note above](https://github.com/mantisbt/mantisbt/pull/1587/files#r355185170)

Reply:
I thought I did reply as part of my last round of comments.
I don't think we should trigger an exception here. It is a valid scenario to have one element in one format that does not have equivalent in another. Mapping routine should not flag that as an error.

Reply:
> I don't think we should trigger an exception here. It is a valid scenario to have one element in one format that does not have equivalent in another. Mapping routine should not flag that as an error.

I agree that it is expected not to have equivalence for all formats, but that does not necessarily make it a valid scenario in this specific context.

My concern is that this may lead to inconsistencies or unexpected behavior when processing these dates, and it may be better to consider these formats as invalid and report them as such, to avoid surprises down the line caused by silently ignoring them.

Again, I have not tested this, so it may not be an actual problem.

Reply:
@dregad I looked into your request and tried to satisfy it but it increasingly looks complex and unnecessary. Can you propose a code change that makes sense to you? I am happy to update the PR with it. If not, I don't think we should block this PR any longer.